src/controllers/resize.ts

```diff
@@ -0,0 +1,62 @@
+import { Context } from 'koa'
+import { Controller, Ctx, Get, Params, Query } from 'amala'
+import { IPFS_TIMEOUT } from '../constants'
+import { Readable } from 'stream'
+import { badRequest, internal, notFound } from '@hapi/boom'
+import { getCachedResize, setCachedResize } from '../helpers/resizedImages'
+import { getContentType } from '../helpers/contentTypes'
+import ipfs, { getDataFromIPFS } from '../helpers/ipfs'
+import sharp from 'sharp'
+
+@Controller('/resize')
+export default class ResizeController {
+  @Get('/image/:cid')
+  async image(
+    @Ctx() ctx: Context,
+    @Params('cid') cid: string,
+    @Query('width') width?: number,
```

Comment:
I wonder if we can use validator for Params, because you can pass something like:
```TS
https://ipfs.sealcred.xyz/resize/image/someRealCid?width=abc&height=abc
```

Reply:
This might be a good idea, because it'll be faster than IPFS looking forever for an image with an invalid CID. I'll try to find regex matcher for CIDs.

Reply:
Added validation for CID here: [516ba06](https://github.com/BigWhaleLabs/obss-ipfs-uploader/pull/12/commits/516ba061c1f68d6129b4aa68493b050ad5b009a5)